rules/Collection22/Rector/CriteriaOrderingConstantsDeprecationRector.php

```diff
@@ -0,0 +1,219 @@
+            return null;
+        }
+
+        if (! $this->criteriaObjectType->isSuperTypeOf(
+            new ObjectType($classConstFetch->class->toCodeString())
+        )->yes()) {
+            return null;
+        }
+
+        return match ($classConstFetch->name->name) {
+            'ASC' => new String_('ASC'),
+            'DESC' => new String_('DESC'),
+        };
+    }
+
+    private function refactorMethodCall(Node\Expr\MethodCall $node): ?Node
+    {
+        if (! $this->isName($node->name, 'orderBy')) {
+            return null;
+        }
+
+        if ($node->isFirstClassCallable()) {
+            return null;
+        }
+
+        $isSuperTypeOfCriteria = $this->criteriaObjectType->isSuperTypeOf($this->nodeTypeResolver->getType($node->var));
+
+        if ($isSuperTypeOfCriteria->no() || $isSuperTypeOfCriteria->maybe()) {
+            return null;
+        }
+
+        $args = $node->getArgs();
+
+        if (count($args) < 1) {
+            return null;
+        }
+
+        $arg = $args[0];
+        if (! $arg instanceof Node\Arg) {
+            return null;
+        }
+
+        if (! $arg->value instanceof Array_) {
+            return null;
+        }
+
+        $nodeHasChange = false;
+        $newItems = [];
+
+        // we parse the array of the first argument
+        foreach ($arg->value->items as $item) {
+
+            if ($item === null) {
+                $newItems[] = $item;
+                continue;
+            }
+
+            if ($item->value instanceof String_ && in_array(
+                $v = strtoupper($item->value->value),
+                ['ASC', 'DESC'],
+                true
+            )) {
+                $newItems[] = $this->buildArrayItem($v, $item->key);
+                $nodeHasChange = true;
+            } elseif (
+                $item->value instanceof Node\Expr\ClassConstFetch
+                && $item->value->class instanceof Node\Name
+                /* @phpstan-ignore-next-line */
+                && $this->criteriaObjectType->isSuperTypeOf(new ObjectType($item->value->class->toCodeString()))
+                && $item->value->name instanceof Node\Identifier
+                && in_array($v = strtoupper((string) $item->value->name), ['ASC', 'DESC'], true)
+            ) {
+                $newItems[] = $this->buildArrayItem($v, $item->key);
+                $nodeHasChange = true;
+            } else {
+                $newItems[] = $item;
+            }
+        }
+
+        if ($nodeHasChange) {
+            return $this->nodeFactory->createMethodCall(
+                $node->var,
+                'orderBy',
+                $this->nodeFactory->createArgs([$this->nodeFactory->createArg(new Array_($newItems))])
+            );
+        }
+
+        return null;
+    }
+
+    /**
+     * @param 'ASC'|'DESC' $direction
+     * @return ArrayItem
+     */
+    private function buildArrayItem(string $direction, Node\Expr|null $key): Node\Expr\ArrayItem
+    {
```

Comment:
$direction need to be transformed to strtoupper first to ensure it always uppercase to compare

Reply:
Actually, this is done on lines [169](https://github.com/rectorphp/rector-doctrine/pull/336/files#diff-8248dc7250edd8eee798927000b8dd81be8511a938eb9a8a2ee6dc44e7f31317R169) and [181](https://github.com/rectorphp/rector-doctrine/pull/336/files#diff-8248dc7250edd8eee798927000b8dd81be8511a938eb9a8a2ee6dc44e7f31317R181). 

But this is maybe not the most readable way for doing this...